pkg/ui/src/views/statements/planView.tsx

```diff
@@ -304,51 +273,96 @@ function PlanNodes(props: {
   );
 }
 
-export function PlanView(props: {
-  title: string,
-  plan: IExplainTreePlanNode,
-}) {
-  const flattenedPlanNodes = flattenTree(props.plan);
+interface PlanViewProps {
+  title: string;
+  plan: IExplainTreePlanNode;
+}
+
+interface PlanViewState {
+  expanded: boolean;
+  showExpandDirections: boolean;
+}
+
+export class PlanView extends React.Component<PlanViewProps, PlanViewState> {
+  constructor(props: PlanViewProps) {
+    super(props);
+    this.state = {
+      expanded: false,
+      showExpandDirections: true,
+    };
+  }
+
+  toggleExpanded = () => {
+    this.setState(state => ({
+      expanded: !state.expanded,
+    }));
+  }
+
+  showExpandDirections() {
+    // Only show directions to show/hide the full plan if content is longer than its max-height.
+    const innerContainer = document.getElementById("plan-view-inner-container");
```

Comment:
Using `document.getElementByID` to grab DOM elements from inside of a React component is discouraged; you're supposed to use refs instead. (i.e. this will break if we ever have two of them on the page, but I guess that's… unlikely). See this guide: https://www.javascriptstuff.com/use-refs-not-ids/

I realize we're under time pressure though. So… idk 😛